### PR TITLE
Add dynamic event system

### PR DESCRIPTION
**Pull request by RobLoach**
Fixes https://github.com/RobLoach/nuklear_console/issues/78, #74, #92, #64

- [ ] Make sure the C++ API is still good
- [x] Remove the macros?

### CODE REVIEW

**Comment by RobLoach** [comment]
Need a way to clean up the one time events

File: nuklear_console.h [hunk 1]
```diff
@@ -662,8 +692,30 @@ NK_API void nk_console_render(nk_console* console) {
...
             }
-            return;
         }
+
+        // Invoke the post-render events.
+        size_t count = cvector_size(console->events);
+        if (count > 0) {
+            nk_bool can_erase = nk_true;
+            for (size_t i = 0; i < count; i++) {
+                // Manage only the post-render events.
```
Comment (RobLoach) [inline nuklear_console.h:702]:
``` suggestion
```

**Comment by RobLoach** [comment]
@brianwatling I think this is looking alright. Mind having a look to see if the C++ API still works?

**Review by brianwatling — APPROVED** [review]
C++ LGTM. Didn't have time to integrate into my code base, but I don't see any reason it won't Just Work. I'll integrate it shortly after you push to master

**Pull request merged**